language-design.md

```diff
@@ -38,6 +96,43 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+There are many ways to define a well-designed language. We believe that 
+language simplicity, understandability and the ability for users to add their 
+own parts encompass a well-designed language. 
+
+I think it would be best to start by using examples from Steele. From a 
+quote we used in question one, Steele explains "if the design of the Java 
+programming language as it is now had been put forth three years ago, it 
+would have failed — of that I am sure" [Steele, 1998]. So it seems like, at 
+least initial, simplicity is a key to good language design. Pavlus states 
+that "He said that their usability testing showed that simply finding 
+natural-language replacements for some of the more abstruse syntax went a 
+long way." [Pavlus, 2012] He agrees with Steele in that he argues that language 
+with simpler syntax, or syntax that the user is more familiar with, allows for 
+new users of the language to accomplish tasks in much less time. This also 
+argues for both understandability and simplicity. Bloch also agrees with 
+understandability by saying that well written APIs should "read like prose." 
+[Bloch, 2006] Steele also argues that a language shouldn't do more than it 
+has to.He says that "what we need is more like a shopping mall, where there 
+are not quite as many choices but most of the goods are well designed and sellers 
+stand up and back what they sell." [Steele, 1998] So a language should allow for 
+flexibility but not do too much for the user. 
+
+We both agreed that badly designed languages boil down to being hard to understand 
+and hard to implement thoughts from our head. This can be either understanding 
+plain syntax or understanding how parts of the language work to having language 
+design that requires work arounds to do tasks that our mind would find simple. 
+
+Pavlus quoted from a researcher that "Perl users were unable to write programs more 
+accurately than those using a language designed by chance." [Pavlus, 2012] This 
+stemmed from, they said, hard to understand syntax. If it is hard for a user to 
+understand the syntax, using that syntax to make difficult programs is going to be 
+even harder. Since languages are designed to make writing programs easier, this is 
+probably a symptom of bad-language design. Bloch agrees by stating a requirement for 
+a good API is "obey the principle of least astonishment." [Bloch, 2006] By having the 
+language do what the user thinks it should do with a specific call, less errors and 
+rewriting will have to occur for the user to get to the right syntax. 
+
```

Comment:
Bad languages have bad syntax and bad usability, but what makes a syntax hard to understand and what makes a language have parts that are difficult to understand? these are harder questions but ones with way more interesting answers